### PR TITLE
Refactor routes

### PR DESCRIPTION
**Pull request by fletch3555**
This is a fairly substantial rewrite of how routes are handled.  I've done some testing locally and all seems to be functioning identical to before, so I don't believe I've introduced any bugs with this.

I'll provide a docs update to go along with this if you're good with it @marktopper and @tnylea

### CODE REVIEW

**Comment by fletch3555** [comment]
This would take care of my needs as part of #35 by allowing developers to import Voyager routes under whatever route configuration they want (subdomain, prefix, etc.)

**Comment by fletch3555** [comment]
@marktopper, Could you take a look at why TravisCI is failing here? As far as I can tell from my testing, this is building the correct route ("voyager.dashboard" = "http://localhost/admin"), but it seems to be seeing the default laravel homepage.

File: publishable/routes/voyager.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+|
+| This file is where you may override any of the routes that are included
+| with Voyager.
+|
+*/
+
+Route::group([
+    'namespace'  => '\\TCG\\Voyager\\Http\\Controllers',
+    'middleware' => 'web',
+    'prefix'     => 'admin',
```
Comment (marktopper) [inline publishable/routes/voyager.php:16]: Prefix is hardcoded now.

File: publishable/database/seeds/MenuItemsTableSeeder.php [hunk 2]
```diff
@@ -46,7 +44,7 @@ public function run()
                 'id'         => 3,
                 'menu_id'    => 2,
                 'title'      => 'Posts',
-                'url'        => "/{$prefix}/posts",
+                'url'        => route('posts.index'),
```
Comment (marktopper) [inline publishable/database/seeds/MenuItemsTableSeeder.php]: Route names in Voyager should be prefixed with Voyager. This seems to be missing in many lines, please go though.

**Comment by marktopper** [comment]
However, I just pulled down your branch, and I myself see the frontpage instead of the dashboard.
I'm simply being redirected.

I recommend that you go though the routes and make sure all of them uses the `voyager` prefix. Then if you still have problems with this, I will look into it.

**Comment by fletch3555** [comment]
I followed the TravisCI script step by step and was still getting the dashboard when I tested. Guess I'll have to retest tomorrow

**Comment by marktopper** [comment]
It seems that Travis does not get the the database seeded correctly.
It leaves out the following tables:
- menu_items
- roles
- settings
- user_roles

Since the seeders are ran in this order:
```
Seeded: CategoriesTableSeeder
Seeded: DataTypesTableSeeder
Seeded: DataRowsTableSeeder
Seeded: UsersTableSeeder
Seeded: PostsTableSeeder
Seeded: PagesTableSeeder
Seeded: MenusTableSeeder
Seeded: MenuItemsTableSeeder
Seeded: SettingsTableSeeder
Seeded: RolesTableSeeder
Seeded: UserRolesTableSeeder
```

I think the problem is in either the `MenusTableSeeder` or `MenuItemsTableSeeder` seeder class.

**Comment by marktopper** [comment]
The issue is in the `MenuItemsTableSeeder`, and are cause that it uses `route(something)`, however the routes for Voyager are not loaded when trying to seed the table, which throws an exception.

**Comment by marktopper** [comment]
I just got it to work by requiring the routes in the installation command.
However this is a bad idea to use `route`, since that will generate a full url like `http://localhost/admin` from `route('voyager.dashboard')` and input it in the database as `slug`.
However then it's not really a slug anymore. What are peoples thoughts on this?

**Comment by fletch3555** [comment]
@marktopper Finally got Travis working!  Not 100% sure I like the solution, but it works.

**Comment by marktopper** [comment]
Nice job. Can you fix the conflicts and let me know, then I will merge into `v0.10`.

**Comment by fletch3555** [comment]
Working on the conflicts now.  This will need a docs update before we tag/release it

**Comment by fletch3555** [comment]
conflicts resolved. waiting on Travis to make sure I didn't break anything

**Pull request merged**